cdk/domino_cdk/eks_stack.py

```diff
@@ -579,6 +580,8 @@ def provision_unmanaged_nodegroup(
                         "--register-with-taints={}".format(",".join(["{}={}".format(k, v) for k, v in taints.items()]))
                     )
                 options["bootstrap_options"] = eks.BootstrapOptions(kubelet_extra_args=" ".join(extra_args))
+            if cfg["ssm_agent"]:
```

Comment:
Don't you want to move it before using user_data in LaunchTemplate to make the logic more obvious?

Reply:
I had the wrong indent on the original version of this you reviewed, but this is meant to only be executed when the code above it is executed.

Reply:
My point is: 
1. You pass a reference.
2. You update an object at the reference later.

This obscures the logic a little bit for the reader who is not aware if user_data is copied to the destination or is left as a reference.

I.e. instead of the code
```
envelope = Envelope()
mailbox.put(envelope)
envelope.letter = "hello"
```

I would write
```
envelope = Envelope()
envelope.letter = "hello"
mailbox.put(envelope)
```